src/history/test/HistoryTests.cpp

```diff
@@ -174,57 +174,158 @@ TEST_CASE("History bucket verification", "[history][catchup]")
+        TestBucketState::FILE_NOT_UPLOADED));
+auto verify = wm.executeWork<DownloadBucketsWork>(
+    buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
 
-        auto verify =
-            wm.executeWork<DownloadBucketsWork>(mBuckets, hashes, *tmpDir);
-        REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+SECTION("hot archive buckets")
+{
+    hotHashes.push_back(bucketGenerator.generateBucket<HotArchiveBucket>(
+        TestBucketState::FILE_NOT_UPLOADED));
+    auto verify = wm.executeWork<DownloadBucketsWork>(
+        buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+    REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+
+SECTION("both live and hot archive buckets")
+{
+    liveHashes.push_back(bucketGenerator.generateBucket<LiveBucket>(
+        TestBucketState::FILE_NOT_UPLOADED));
+    hotHashes.push_back(bucketGenerator.generateBucket<HotArchiveBucket>(
+        TestBucketState::FILE_NOT_UPLOADED));
+    auto verify = wm.executeWork<DownloadBucketsWork>(
+        buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+    REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+#endif
+}
+SECTION("download succeeds but unzip fails"){SECTION("live buckets"){
+    liveHashes.push_back(bucketGenerator.generateBucket<LiveBucket>(
+        TestBucketState::CORRUPTED_ZIPPED_FILE));
+auto verify = wm.executeWork<DownloadBucketsWork>(
+    buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+SECTION("hot archive buckets")
+{
+    hotHashes.push_back(bucketGenerator.generateBucket<HotArchiveBucket>(
+        TestBucketState::CORRUPTED_ZIPPED_FILE));
+    auto verify = wm.executeWork<DownloadBucketsWork>(
+        buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+    REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+
+SECTION("both live and hot archive buckets")
+{
+    liveHashes.push_back(bucketGenerator.generateBucket<LiveBucket>(
+        TestBucketState::CORRUPTED_ZIPPED_FILE));
+    hotHashes.push_back(bucketGenerator.generateBucket<HotArchiveBucket>(
+        TestBucketState::CORRUPTED_ZIPPED_FILE));
+    auto verify = wm.executeWork<DownloadBucketsWork>(
+        buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+    REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+#endif
+}
+SECTION("verify fails hash mismatch"){SECTION("live buckets"){
+    liveHashes.push_back(bucketGenerator.generateBucket<LiveBucket>(
+        TestBucketState::HASH_MISMATCH));
+auto verify = wm.executeWork<DownloadBucketsWork>(
+    buckets, hotBuckets, liveHashes, hotHashes, *tmpDir);
+REQUIRE(verify->getState() == BasicWork::State::WORK_FAILURE);
+}
+
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
```

Comment:
Does this need to be compile-time-gated?

Reply:
Yes, reason being this test manually calls Hot Archive functions that are otherwise protocol gated. Without the ifdef, we call these in p22 curr, which fails asserts.